bigquery/google/cloud/bigquery/job.py

```diff
@@ -572,12 +588,20 @@ def schema(self, value):
         :type value: list of :class:`SchemaField`
         :param value: fields describing the schema
 
-        :raises: TypeError if 'value' is not a sequence, or ValueError if
-                 any item in the sequence is not a SchemaField
+        :raises TypeError: If ``value`is not a sequence.
+        :raises ValueError: If any item in the sequence is not
+                            a ``SchemaField``.
         """
-        if not all(isinstance(field, SchemaField) for field in value):
-            raise ValueError('Schema items must be fields')
-        self._schema = tuple(value)
+        if not value:
+            self._schema = ()
+        else:
+            if not all(isinstance(field, SchemaField) for field in value):
+                raise ValueError('Schema items must be fields')
+            if self._configuration is not None and self.autodetect:
```

Comment:
Hi @dhermes, thank you for the review!

Indeed I used the validation of `self._configuration` being already set because otherwise during instantiation of the `LoadTableFromStorageJob` class it would follow straight to the validation of `if self.autodetect: ...` which was not created yet (since `self._configuration = _LoadingConfiguration` was called after `self.schema = shema`).

If `self._configuration` is called before `self.schema` then it's indeed not necessary (I've chosen to do like I did so to not create restrictions on the order the variables are created, maybe it's easier to maintain a code with less restrictions, but as you have documented about `self.schema` having to occur after all variables were loaded then I think it's all fine!).

Leaving `self._configuration` be loaded before schema makes its validation unnecessary indeed, it could be just `if self.autodetect: raise ValueError()`